### PR TITLE
[DataProtectionBackup] Polish the API to prepare for release

### PR DESCRIPTION
**Pull request by Yao725**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/Customized/Models/BackupInstancePolicyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        /// <summary>
+        /// Gets or sets the DataStore Parameters
+        /// Please note <see cref="DataStoreSettings"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+        /// The available derived classes include <see cref="OperationalDataStoreSettings"/>.
+        /// Please note <see cref="DataStoreSettings"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+        /// The available derived classes include <see cref="OperationalDataStoreSettings"/>.
+        /// </summary>
+        [Obsolete("DataStoreParametersList is obsolete and will be removed in a future release. Please do not use it any longer.", false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public IList<DataStoreSettings> DataStoreParametersList
```
Comment (Yao725) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/Customized/Models/BackupInstancePolicyInfo.cs:25]: When polishing the API, I realized that we might have an issue in single property flatten for collection type, created [an issue](https://github.com/Azure/autorest.csharp/issues/3058) to track.

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public static partial class DataProtectionBackupExtensions
...
         public static Azure.AsyncPageable<Azure.ResourceManager.DataProtectionBackup.DataProtectionBackupVaultResource> GetDataProtectionBackupVaultsAsync(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public static Azure.ResourceManager.DataProtectionBackup.DeletedBackupInstanceResource GetDeletedBackupInstanceResource(this Azure.ResourceManager.ArmClient client, Azure.Core.ResourceIdentifier id) { throw null; }
+        public static Azure.ResourceManager.DataProtectionBackup.DataProtectionDeletedBackupInstanceResource GetDataProtectionDeletedBackupInstanceResource(this Azure.ResourceManager.ArmClient client, Azure.Core.ResourceIdentifier id) { throw null; }
         public static Azure.Response<Azure.ResourceManager.DataProtectionBackup.ResourceGuardResource> GetResourceGuard(this Azure.ResourceManager.Resources.ResourceGroupResource resourceGroupResource, string resourceGuardsName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:20]: should we find a more descriptive name for this resource? This name looks quite generic

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/autorest.md [hunk 3]
```diff
@@ -218,6 +220,27 @@ rename-mapping:
   SyncType: BackupInstanceSyncType
   ScheduleBasedBackupCriteria.daysOfTheWeek: DaysOfWeek
   ScheduleBasedBackupCriteria.weeksOfTheMonth: WeeksOfMonth
+  DeletedBackupInstanceResource: DataProtectionDeletedBackupInstance
```
Comment (ArthurMa1978) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/autorest.md:223]: Is `DeletedDataProtection`BackupInstance` better?

**Review by ArthurMa1978 — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs (hunk 2 above)
Comment (Yao725) [reply sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:20]: Do you mean `ResourceGroupResource`? Unfortunately, we can't. This resource existed at the time of the last stable release.

**Review by archerzz — APPROVED** [review]

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/autorest.md (hunk 3 above)
Comment (Yao725) [reply sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/autorest.md]: Good point, fixed!

**Pull request merged**